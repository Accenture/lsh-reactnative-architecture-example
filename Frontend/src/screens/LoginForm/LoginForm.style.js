const loginFormStyles = {
  errorTextStyle: {
    fontSize: 20,
    alignSelf: 'center',
    color: 'red',
    padding: 2,
  },
  translateText: {
    fontSize: 15,
    color: '#007aff',
  },
  translateContainer: {
    display: 'flex',
    padding: 15,
    marginLeft: 130,
    marginTop: 20,
    flexDirection: 'row',
  },
};
export default loginFormStyles;
