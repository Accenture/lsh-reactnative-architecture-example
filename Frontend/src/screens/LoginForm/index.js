import React, { Component } from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { observer, inject, PropTypes } from 'mobx-react';
import Card from 'Frontend/src/components/Card';
import CardSection from 'Frontend/src/components/CardSection';
import Button from 'Frontend/src/components/Button';
import LoginInput from 'Frontend/src/components/LoginInput';
import Loader from 'Frontend/src/components/Loader';
import translate from 'Frontend/src/commons/Language';
import Layout from 'Frontend/src/layout';
import styles from './LoginForm.style';

@inject('loginStore', 'languageStore')
@observer
export default class LoginForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
    };
    this.saveEmail = this.saveEmail.bind(this);
    this.savePassword = this.savePassword.bind(this);
    this.onButtonPress = this.onButtonPress.bind(this);
    this.renderButton = this.renderButton.bind(this);
  }

  onLoginSuccess() {
    this.setState({ loading: false });
    this.props.loginStore.setErrorMessage('');
    this.props.navigation.navigate('home');
  }

  onLoginFail() {
    this.props.loginStore.setErrorMessage(translate('AUTH_error'));
    this.setState({ loading: false });
  }

  saveEmail(e) {
    this.props.loginStore.setEmail(e.nativeEvent.text);
  }

  savePassword(e) {
    this.props.loginStore.setPassword(e.nativeEvent.text);
  }

  onButtonPress() {
    this.setState({ loading: true });
    const loggedIn =
      this.props.loginStore.requestLogInWithEmailPassword &&
      this.props.loginStore.isLoggedIn;
    if (loggedIn) {
      this.onLoginSuccess();
    } else {
      this.onLoginFail();
    }
  }

  renderButton() {
    if (this.state.loading) {
      return <Loader size="small" />;
    }
    return (
      <Button onPress={this.onButtonPress}>
        <Text>{translate('AUTH_login')}</Text>
      </Button>
    );
  }

  render() {
    const { errorTextStyle, translateText, translateContainer } = styles;
    const langs = ['fi', 'en', 'sv'];
    return (
      <Layout headerText={'AUTH_title'} featureName="Auth" hideDrawer>
        <View key={this.props.languageStore.key}>
          <Card>
            <CardSection>
              <LoginInput
                placeholder={'email@email.com'}
                label={translate('AUTH_email')}
                value={this.props.loginStore.email}
                onChangeText={this.saveEmail}
              />
            </CardSection>
            <CardSection>
              <LoginInput
                secureTextEntry
                placeholder={'*******'}
                label={translate('AUTH_password')}
                value={this.props.loginStore.password}
                onChangeText={this.savePassword}
              />
            </CardSection>
            <Text style={errorTextStyle}>
              {this.props.loginStore.errorMessage}
            </Text>
            <CardSection>{this.renderButton()}</CardSection>
          </Card>
          <View style={translateContainer}>
            {langs.map(lang => (
              <TouchableOpacity
                key={lang}
                onPress={() =>
                  this.props.languageStore.setCurrentLanguage(lang)
                }
              >
                <Text style={translateText}> {lang.toUpperCase()} </Text>
              </TouchableOpacity>
            ))}
          </View>
        </View>
      </Layout>
    );
  }
}

LoginForm.propTypes = {
  loginStore: PropTypes.objectOrObservableObject,
  navigation: PropTypes.objectOrObservableObject,
  languageStore: PropTypes.objectOrObservableObject,
};
