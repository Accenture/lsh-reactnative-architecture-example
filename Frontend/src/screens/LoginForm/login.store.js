// @flow

import { observable, action, computed } from 'mobx';
import { AsyncStorage } from 'react-native';

export default class LoginStore {
  @observable
  email = '';

  @observable
  password = '';

  @observable
  jwtToken = '';

  @observable
  isAuthenticated = false;

  @observable
  errorMessage = '';

  @action
  setEmail(newEmail: string) {
    this.email = newEmail;
  }

  @action
  setPassword(newPassword: string) {
    this.password = newPassword;
  }

  @action
  async setJwtToken(newToken: string) {
    this.jwtToken = newToken;
    await AsyncStorage.removeItem(this.jwtToken);
    await AsyncStorage.setItem(this.jwtToken, newToken);
  }

  @action
  setErrorMessage(message: string) {
    this.errorMessage = message;
  }

  async getJwtToken() {
    if (this.jwtToken) {
      return (this.jwtToken: string);
    }
    await AsyncStorage.getItem(this.jwtToken);
    return this.jwtToken;
  }

  @computed
  get isLoggedIn(): boolean {
    return !!this.jwtToken && !!this.isAuthenticated;
  }

  @computed
  get requestLogInWithEmailPassword(): boolean {
    if (this.email === 'email' && this.password === 'password') {
      this.jwtToken = 'newToken';
      this.isAuthenticated = true;
      return this.isAuthenticated;
    }
    return false;
  }

  @action
  logout() {
    this.jwtToken = '';
    this.isAuthenticated = false;
  }
}
