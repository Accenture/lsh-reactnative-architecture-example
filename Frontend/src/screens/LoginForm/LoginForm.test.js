import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './index';

describe('Tests login form component', () => {
  const store = {
    loginStore: {
      email: '',
      password: '',
      jwtToken: '',
    },
    languageStore: {
      key: 1,
    },
  };
  it('matches snapshot', () => {
    const comp = shallow(<LoginForm.wrappedComponent {...store} />);
    expect(comp).toMatchSnapshot();
  });
});
