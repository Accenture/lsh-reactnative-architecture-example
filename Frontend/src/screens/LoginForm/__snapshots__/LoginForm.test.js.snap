// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests login form component matches snapshot 1`] = `
<inject-Layout-with-featureToggleStore
  featureName="Auth"
  headerText="AUTH_title"
  hideDrawer={true}
  onPress={[Function]}
>
  <Component
    key="1"
  >
    <Card>
      <CardSection>
        <Input
          label="Email"
          onChangeText={[Function]}
          placeholder="email@email.com"
          secureTextEntry={false}
          value=""
        />
      </CardSection>
      <CardSection>
        <Input
          label="Password"
          onChangeText={[Function]}
          placeholder="*******"
          secureTextEntry={true}
          value=""
        />
      </CardSection>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "red",
            "fontSize": 20,
            "padding": 2,
          }
        }
      />
      <CardSection>
        <Button
          onPress={[Function]}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Log In
          </Text>
        </Button>
      </CardSection>
    </Card>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "marginLeft": 130,
          "marginTop": 20,
          "padding": 15,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        key="fi"
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#007aff",
              "fontSize": 15,
            }
          }
        >
           
          FI
           
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="en"
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#007aff",
              "fontSize": 15,
            }
          }
        >
           
          EN
           
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        activeOpacity={0.2}
        key="sv"
        onPress={[Function]}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#007aff",
              "fontSize": 15,
            }
          }
        >
           
          SV
           
        </Text>
      </TouchableOpacity>
    </Component>
  </Component>
</inject-Layout-with-featureToggleStore>
`;
